### PR TITLE
Update range addressing (meta) records on replica changes.

### PR DESCRIPTION
**Pull request by spencerkimball**
Renamed storage/db.go to storage/addressing.go and augmented ReplicaReplica
unittest to check that addressing intents are resolved and range descriptor
copied.

### CODE REVIEW

File: storage/addressing.go [hunk 1]
```diff
@@ -67,6 +67,17 @@ func MergeRangeAddressing(left, merged *proto.RangeDescriptor) ([]client.Call, e
 	return calls, nil
 }
 
+// UpdateRangeAddressing overwrites the meta1 and meta2 range addressing
+// records for the descriptor.
```
Comment (tschottdorf) [inline storage/addressing.go]: Add a comment about the calls returned.

Comment (spencerkimball) [inline storage/addressing.go]: Done.

File: storage/addressing.go [hunk 2]
```diff
@@ -67,6 +67,17 @@ func MergeRangeAddressing(left, merged *proto.RangeDescriptor) ([]client.Call, e
...
+func UpdateRangeAddressing(desc *proto.RangeDescriptor) ([]client.Call, error) {
+	var calls []client.Call
+	var err error
+	if calls, err = updateRangeAddressing(calls, desc, putMeta); err != nil {
```
Comment (bdarnell) [inline storage/addressing.go]: This whole function could be replaced with `return updateRangeAddressing(calls, desc, putMeta)` (unless you're deliberately trying to ensure that one or the other of calls and err are nil, in which case there should be a comment to that effect).

File: storage/addressing.go [hunk 3]
```diff
@@ -67,6 +67,18 @@ func MergeRangeAddressing(left, merged *proto.RangeDescriptor) ([]client.Call, e
 	return calls, nil
 }
 
+// UpdateRangeAddressing overwrites the meta1 and meta2 range addressing
+// records for the descriptor. Returns a slice of calls necessary to
+// update the records on the KV database.
+func UpdateRangeAddressing(desc *proto.RangeDescriptor) ([]client.Call, error) {
```
Comment (bdarnell) [inline storage/addressing.go]: Why is this exported? The only use is in this package.

**Comment by bdarnell** [comment]
LGTM

File: storage/addressing.go (hunk 3 above)
Comment (spencerkimball) [inline storage/addressing.go]: Unexported it.

File: storage/client_test.go [hunk 4]
```diff
@@ -379,178 +373,3 @@ func internalTruncateLogArgs(index uint64, raftID int64, storeID proto.StoreID)
...
-				}
-			}
-		}
-	}
-}
-
-// TestUpdateRangeAddressingSplitMeta1 verifies that it's an error to
-// attempt to update range addressing records that would allow a split
-// of meta1 records.
-func TestUpdateRangeAddressingSplitMeta1(t *testing.T) {
```
Comment (bdarnell) [inline storage/client_test.go]: Do we have other tests that cover what these tests did? If these tests were useful we should keep them even if it means some extra exported functions.

**Pull request merged**